monkey/infection_monkey/monkey.py

```diff
@@ -394,9 +395,9 @@ def cleanup(self):
 
             reset_signal_handlers()
 
-            if self._inbound_tunnel_opened:
-                self._monkey_inbound_tunnel.stop()
-                self._monkey_inbound_tunnel.join()
+            if self._relay and self._relay.is_alive():
+                self._relay.stop()
+                self._relay.join()
```

Comment:
Let's set a timeout on this `join()`. Maybe 60 seconds? I know there are other timeouts in play, but let's not hang the whole agent up if the relay won't shut down.

We want the agent to shut down in 120 seconds or less from the time the user hits the stop button in the UI.